examples/contrib/readhar.py

```diff
@@ -0,0 +1,134 @@
+"""Reads HAR files into flow objects"""
+import asyncio
+import codecs
+import json
+import logging
+import time
+
+from mitmproxy import command
+from mitmproxy import connection
+from mitmproxy import ctx
+from mitmproxy import exceptions
+from mitmproxy import http
+from mitmproxy import types
+
+
+class ReadHar:
+    def __init__(self):
+        self.harFile = {}
+        self.flows = []
+
+    def fix_headers(self, request_headers):
+        """Changes provided headers into flow-friendly format"""
+        flow_headers = []
+        for header in request_headers:
+            if type(header) == dict and header.get("name", None):
+                key = header["name"]
+                value = header["value"]
+            else:
+                key = header[0]
+                value = header[1]
+            flow_headers.append((bytes(key, "utf-8"), bytes(value, "utf-8")))
+
+        return flow_headers
+
+    def request_to_flow(self, request_json):
+        """
+        Creates a HTTPFlow object from a given entry in HAR file
+        """
+
+        requestMethod = request_json["request"].get("method", "")
```

Comment:
Are there cases where "method" is not present? If yes, let's document that (see comment above). If no, we should probably just do `request_json["request"]["method"]`.